src/test/java/org/junit/tests/running/classes/ThreadsTest.java

```diff
@@ -0,0 +1,98 @@
+package org.junit.tests.running.classes;
+
+import org.junit.AfterClass;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.internal.runners.ErrorReportingRunner;
+import org.junit.runner.Description;
+import org.junit.runner.JUnitCore;
+import org.junit.runner.Request;
+import org.junit.runner.Result;
+import org.junit.runner.RunWith;
+import org.junit.runner.Runner;
+import org.junit.runner.notification.RunListener;
+import org.junit.runners.BlockJUnit4ClassRunner;
+import org.junit.runners.model.InitializationError;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.CopyOnWriteArrayList;
+
+import static java.util.Arrays.asList;
+import static java.util.Collections.singletonList;
+import static org.junit.Assert.assertEquals;
+
+public class ThreadsTest {
+
+    private List<Boolean> interruptedFlags = new CopyOnWriteArrayList<Boolean>();
+    private JUnitCore core = new JUnitCore();
+
+    public static class TestWithInterrupt {
+
+        @Test
+        public void interruptCurrentThread() {
+            Thread.currentThread().interrupt();
+        }
+
+        @Test
+        public void otherTestCaseInterruptingCurrentThread() {
+            Thread.currentThread().interrupt();
+        }
+
+    }
+
+    @Test
+    public void currentThreadInterruptedStatusIsClearedAfterEachTestExecution() {
+        core.addListener(new RunListener() {
+            @Override
+            public void testFinished(Description description) {
+                interruptedFlags.add(Thread.currentThread().isInterrupted());
```

Comment:
1) This listener is duplicated in both tests. As JUnitCore is a field, might be add this listener from a `@Before` method?
2) Another thing this listener can do is just `assertFalse()`, WDYT?

Reply:
The listeners actually implement a different method but have the same implementation. Just doing `assertFalse()` would have the downside that we're not checking that the listener was actually called.